astroquery/mast/collections.py

```diff
@@ -241,20 +245,13 @@ def query_criteria_async(self, catalog, pagesize=None, page=None, **criteria):
         objectname = criteria.pop('objectname', None)
         radius = criteria.pop('radius', 0.2*u.deg)
 
-        if objectname and coordinates:
-            raise InvalidQueryError("Only one of objectname and coordinates may be specified.")
-
-        if objectname:
-            coordinates = self.resolve_object(objectname)
-
-        if coordinates:
-            # Put coordinates and radius into consitant format
-            coordinates = commons.parse_coordinates(coordinates)
+        if objectname or coordinates:
+            coordinates = utils._parse_input_location(coordinates, objectname)
```

Comment:
I don't think a public utility method shouldn't be named with "_".

Reply:
I wasn't thinking of it as a public method. Do you think it should be public?

Reply:
Will handle in future PR.